irods/connection.py

```diff
@@ -273,17 +274,19 @@ def _connect(self):
         return version_msg.get_main_message(VersionResponse)
 
     def disconnect(self):
-        disconnect_msg = iRODSMessage(msg_type='RODS_DISCONNECT')
-        self.send(disconnect_msg)
-        try:
-            # SSL shutdown handshake
-            self.socket = self.socket.unwrap()
-        except AttributeError:
-            pass
-        self.socket.shutdown(socket.SHUT_RDWR)
-        self.socket.close()
-        self.socket = None
-        self._disconnected = True
+        # fileno() returns -1 for a closed socket
```

Comment:
New condition for calling disconnect. Only call disconnect() if fileno is not -1 (as the socket is already closed).

Reply:
The condition in `__del__` looked like this...
```python
if self.socket and getattr(self,"_disconnected",False):
```
The second condition (which is equivalent to `getattr(self,"_disconnected",False) == True`) has been reversed here. This seems more correct from my naive perspective. @d-w-moore, does that seem right?